File: find-peak-element/find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            if((i==0||nums[i]>nums[i-1])&&(i==nums.size()-1||nums[i]>nums[i+1]))
                return i;
        };
        return -1;
    }
};

File: find-peak-element/find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int max=Integer.MIN_VALUE;
      TreeMap<Integer,Integer>map=new TreeMap<>();
      for(int i=0;i<nums.length;i++)
      {
         if(!map.containsKey(nums[i]))
         {
           max=Math.max(max,nums[i]);
           map.put(nums[i],i);
         }
        
      }
      return map.get(max);
    
    }
}